tests/page/expect-misc.spec.ts

```diff
@@ -220,6 +220,33 @@ test.describe('toHaveClass', () => {
     const error = await expect(locator).toHaveClass(['foo', 'bar', /[a-z]az/], { timeout: 1000 }).catch(e => e);
     expect(error.message).toContain('expect.toHaveClass with timeout 1000ms');
   });
+
+  test('allow matching partial class names', async ({ page }) => {
+    await page.setContent('<div class="foo bar"></div>');
+    const locator = page.locator('div');
+    await expect(locator).toHaveClass('foo', { partial: true });
+    await expect(locator).toHaveClass('bar', { partial: true });
+    await expect(
+        expect(locator).toHaveClass(/f.o/, { partial: true })
+    ).rejects.toThrow('Partial matching does not support regular expressions. Please provide a string value.');
+    await expect(locator).not.toHaveClass('foo');
+    await expect(locator).not.toHaveClass('foo', { partial: false });
+    await expect(locator).toHaveClass('  bar   foo ', { partial: true });
```

Comment:
This test does not check that you can list more then one, but not all classes present on the element.